CDCgov/prime-reportstream
# 8588 - Show filtered values in Daily Data date filter and other misc updates

This PR ...

- Renamed files from deliveries to Daily data (more explicit)
- fixed date filter to keep date being searched upon
- fixed clear button not working


Test Steps:
1. Log into RS
2. Navigate to Daily Data
3. Enter a date and notice that it remains upon loading the table data.
4. Click "Clear" and all filter values should default to original.

## Changes
![Screenshot 2023-12-28 at 11 45 08 AM](https://github.com/CDCgov/prime-reportstream/assets/102491809/b0ad9a5e-5ab6-47a1-bbde-6102f7b53cb2)


## Checklist

### Testing


## Linked Issues
- Fixes #8588
